game/src/main/scala/io/aigar/game/Virus.scala

```diff
@@ -0,0 +1,48 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace.Vector2
+import io.aigar.game.serializable.Position
+
+object Virus {
+  final val Quantity = 15
+  final val Mass = 100
+}
+
+class Virus(var position: Vector2) {
+
+  def update(grid: Grid, players: List[Player]): Unit = {
+    if(detectCollisions(players)) {
+      // TODO Add the virus consumption into the cell
```

Comment:
You are right! I've just realized what you mean, so when there's a collision, I'll return the cell, and when there's not, I'll return nothing, right?